### ISSUE TITLE
blendMode=”multiply” in Nifty renders incorrectly

### ISSUE THREAD

**Issue by ghost** [jMonkeyEngine/jmonkeyengine#99]
Reference on googlecode: https://code.google.com/p/jmonkeyengine/issues/detail?id=609

**Comment by stephengold** [comment]
5 years later, I'm still working around this issue.

**Comment by stephengold** [comment]
Forum discussion: https://hub.jmonkeyengine.org/t/nifty-labels-inside-a-scrollpanel-control/27402

**Comment by stephengold** [comment]
With the help of the Internet Archive, I re-created the workaround from 2014:
   https://hub.jmonkeyengine.org/t/text-within-a-nifty-scroll-panel-acquires-a-black-surround/41768

**Comment by stephengold** [comment]
I discovered a possible fix for this issue: replace 
```
    } else if (blendMode == BlendMode.MULIPLY) {
        return RenderState.BlendMode.Modulate;
```
with
```
    } else if (blendMode == BlendMode.MULIPLY) {
        return RenderState.BlendMode.Alpha;
```
in both `JmeBatchRenderBackend.java` and `RenderDeviceJme.java`.

I imagine such a change might have undesirable side effects, but so far I haven't observed any.

**Comment by tonihele** [comment]
Ok, I tested this. I don't know what to make of these results. I took our custom styling off from the ScrollPanel. I typed the JME BlendMode on the result images. I could imagine that the Nifty's multiply translates to JME's Modulate. We should have some sort of basic test case.
For our part. I didn't see any ill-effects either way. But we don't seem to use the blend mode multiply at all by default.
https://github.com/nifty-gui/nifty-gui/wiki/Effects#blendMode
![image](https://user-images.githubusercontent.com/8344766/57016496-23b17b80-6c23-11e9-8e43-f0c792e0ee20.png)
![image](https://user-images.githubusercontent.com/8344766/57016504-2ad88980-6c23-11e9-810a-503e005e5377.png)

**Comment by stephengold** [comment]
Thanks for trying this. So your custom scroll panel doesn't use blendMode="multiply"? I'm wondering if it has any application outside of scroll panels...

**Comment by tonihele** [comment]
Well, the in the pictures it uses. I dropped our custom styling, so I assume it used. And changing the value it translated to in JME has the following outcomes. But I don't know what to make of that.

I guess the attribute can be used in effects etc.?

**Comment by stephengold** [comment]
I've decided the only available fix is too risky for JME 3.2.4

**Comment by stephengold** [comment]
Fixed in `master` branch by Pr #1082.
